### DISCUSSION TITLE
UR control via Python ros2 node

### DISCUSSION THREAD

**Discussion by Icon45** [UniversalRobots/Universal_Robots_ROS2_Driver#699] [Q&A]
Hi everyone,

I'm trying to control the joints of an ur10 (in Gazebo) via a node. I construct a FollowJointTrajectory.Goal() message and I'm trying to send that via an ActionClient(self, FollowJointTrajectory, "joint_trajectory_controller). 

Unfortunately I get stuck when I'm waiting for the server, which I thought wis being spawned when launching the UR. Does anyone know what's going on and wants to help me?

**Comment by bmaxdk** [comment]
Can you see the action server?
```bash
$ ros2 action list
```
If action server name is shown as you provided by the UR10's joint trajotry controller, the name include the namespace such as `/ur10/joint_trajectory_controller/follow_joint_trajectory` should be provided.

For python for action server should be similar to:
```python
import actionlib
from control_msgs.msg import FollowJointTrajectory, FollowJointTrajectoryGoal

client = actionlib.SimpleActionClient('joint_trajectory_controller/follow_joint_trajectory', FollowJointTrajectory)
client.wait_for_server()

goal = FollowJointTrajectoryGoal()
client.send_goal(goal)
client.wait_for_result()
```
